packages/semcom-node/lib/store/services/pod-sync.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Handler, TypedKeyValueStore } from '@digita-ai/handlersjs-core';
+import { Parser } from 'n3';
+import { from, Observable, of } from 'rxjs';
+
+export class PodSyncService<S extends string, M extends { [s in S]: string[] }>
+  extends Handler<void, void> {
+
+  /**
+   *
+   * @param storage key in which the storage is located
+   * @param store the given store
+   */
+  constructor(
+    private readonly storage: S,
+    private readonly store: TypedKeyValueStore<M>,
+    private readonly localPod: string,
+  ) { super(); }
+
+  canHandle(): Observable<boolean> {
+
+    return of(true);
+
+  }
+
+  private async componentsInPod(uri: string): Promise<Set<string>> {
```

Comment:
When using async functions, prefer `await` over `then` notation. That should also make a normal try-catch block possible here, instead of convoluted new promises.